lib/ex_force/client/client.ex

```diff
@@ -0,0 +1,41 @@
+defmodule ExForce.Client do
+  @moduledoc """
+  HTTP Client for Salesforce REST API
+
+  ## Adapter
+
+  Defaults to ExForce.Client.Tesla. To use your own adapter, set it via Mix configuration.
+
+  ```elixir
+  config :ex_force, client: ClientMock
+  ```
+  """
+
+  alias ExForce.{
+    Request,
+    Response
+  }
+
+  @type client :: any()
```

Comment:
Usually `t` in a module defining a struct represents the type of the struct. So let's not add another type (`ExForce.Client.client`) here. `ExForce.Client.t` is enough.

The common pattern is, add this to the top-level module (E.g. `ExForce.client` type)

Reply:
Fixed with dc288e04fd2dc5e95931f73bc9de9a4acf871ea1

Also moved file to lib/ex_force directory in 4e416920451217efee3ef27224287dab72708c53 in order to follow module naming